controllers/cluster_controller.go

```diff
@@ -335,6 +335,7 @@ func (r *ClusterReconciler) getDnsServiceForPublicRecords(ctx context.Context, l
 			SubscriptionID: r.BaseZoneSubscriptionID,
 			TenantID:       r.BaseZoneTenantID,
 		},
+		ResourceTags: infracluster.GetResourceTagsFromInfraClusterAnnotations(clusterScope.InfraClusterAnnotations()),
```

Comment:
We set the resource tags inside the dns scope based on the infracluster annotations